google-cloud-spanner-hibernate-dialect/src/main/java/com/google/cloud/spanner/hibernate/schema/TableDependencyTracker.java

```diff
@@ -54,8 +57,17 @@ public void initializeDependencies(Metadata metadata, Action schemaAction) {
     HashMap<Table, Table> dependencies = new HashMap<>();
 
     for (Table childTable : metadata.collectTableMappings()) {
-      Interleaved interleaved = SchemaUtils.getInterleaveAnnotation(childTable, metadata);
-      if (interleaved != null) {
+      Interleaved interleaved;
+      Class<?> entity = SchemaUtils.getEntityClass(childTable, metadata);
```

Comment:
Was there a reason why the previous way of `SchemaUtils.getInterleaveAnnotation(childTable, metadata)` didn't work? I think it was simpler since it avoids the `interleaved = xxx` in the if statement.

Reply:
We also need the reference to the class that is annotated with `@Interleaved` when doing the validation, not just the annotation itself. I could return a `pair<Class, Interleaved>`, or even just remove the inline `interleaved = xxx` and grab the annotation from the class twice if its clearer though?

Reply:
Ah I see. Thanks for the change, this is great!